HPI Plugin 3.58 fails core build on Maven ≤ 3.9.5

### Jenkins and plugins versions report

n/a

### What Operating System are you using (both controller, and any agents involved in the problem)?

n/a

### Reproduction steps

1. Put Maven 3.9.5 on the PATH
2. Check out jenkinsci/jenkins
3. mvn clean install -Pquick-build

### Expected Results

It works

### Actual Results

```
[ERROR] Failed to execute goal org.jenkins-ci.tools:maven-hpi-plugin:3.58-SNAPSHOT:record-core-location (default) on project jenkins-core: Execution default of goal org.jenkins-ci.tools:maven-hpi-plugin:3.58-SNAPSHOT:record-core-location failed: Unable to load the mojo 'record-core-location' (or one of its required components) from the plugin 'org.jenkins-ci.tools:maven-hpi-plugin:3.58-SNAPSHOT': com.google.inject.ProvisionException: Unable to provision, see the following errors:
[ERROR] 
[ERROR] 1) No implementation for PluginWorkspaceMap was bound.
[ERROR]   while locating RecordCoreLocationMojo
[ERROR]   at ClassRealm[plugin>org.jenkins-ci.tools:maven-hpi-plugin:3.58-SNAPSHOT, parent: ClassLoaders$AppClassLoader@42110406]
[ERROR]       \_ installed by: WireModule -> PlexusBindingModule
[ERROR]   while locating Mojo annotated with @Named("org.jenkins-ci.tools:maven-hpi-plugin:3.58-SNAPSHOT:record-core-location")
[ERROR] 
[ERROR] 1 error
[ERROR] 
[ERROR] ======================
[ERROR] Full classname legend:
[ERROR] ======================
[ERROR] ClassLoaders$AppClassLoader: "jdk.internal.loader.ClassLoaders$AppClassLoader"
[ERROR] Mojo:                        "org.apache.maven.plugin.Mojo"
[ERROR] Named:                       "com.google.inject.name.Named"
[ERROR] PlexusBindingModule:         "org.eclipse.sisu.plexus.PlexusBindingModule"
[ERROR] PluginWorkspaceMap:          "org.jenkinsci.maven.plugins.hpi.PluginWorkspaceMap"
[ERROR] RecordCoreLocationMojo:      "org.jenkinsci.maven.plugins.hpi.RecordCoreLocationMojo"
[ERROR] WireModule:                  "org.eclipse.sisu.wire.WireModule"
[ERROR] ========================
[ERROR] End of classname legend:
[ERROR] ========================
[ERROR] 
[ERROR]       role: org.apache.maven.plugin.Mojo
[ERROR]   roleHint: org.jenkins-ci.tools:maven-hpi-plugin:3.58-SNAPSHOT:record-core-location
[ERROR] -> [Help 1]
```

### Anything else?

Bisected to https://github.com/jenkinsci/maven-hpi-plugin/pull/652

### Are you interested in contributing a fix?

_No response_

--- a/pom.xml
+++ b/pom.xml
@@ -46,7 +46,7 @@
     <gitHubRepo>jenkinsci/${project.artifactId}</gitHubRepo>
     <jetty.version>12.0.13</jetty.version>
     <!-- if we update this all consumers need to be running at least this version -->
-    <maven.version>3.8.1</maven.version>
+    <maven.version>3.9.6</maven.version>
     <maven-plugin-tools.version>3.15.0</maven-plugin-tools.version>
     <maven-resolver.version>2.0.1</maven-resolver.version>
     <spotless.check.skip>false</spotless.check.skip>

